manifests/tools.pp

```diff
@@ -0,0 +1,206 @@
+# @param backup_and_sync_enable
+#   Whether or not backup_and_sync script is automated
+#
+# @param grafana_admin_password
+#   Password of Grafana admin user
+# @param grafana_admin_user
+#   Username of Grafana admin user
+#
+# @param ldap_sync_base_dir
+#   Base directory used by ldap sync
+# @param ldap_sync_cron_hour
+#   hour that the ldap_sync cron should run
+# @param ldap_sync_cron_minute
+#   minute that the ldap_sync cron should run
+# @param ldap_sync_cron_month
+#   month that the ldap_sync cron should run
+# @param ldap_sync_cron_monthday
+#   monthday that the ldap_sync cron should run
+# @param ldap_sync_cron_weekday
+#   weekday that the ldap_sync cron should run
+# @param ldap_sync_enable
+#   Whether or not ldap_sync script is automated
+# @param ldap_sync_groups
+#   List of ldap groups to sync to Grafana
+#
+# @param whitelabel_subtitle
+#   Custom subtitle applied to Grafana instance
+# @param whitelabel_title
+#   Custom title applied to Grafana instance
+#   
+# @example
+#   include profile_metrics::tools
+class profile_metrics::tools (
+  String $backup_and_sync_cron_hour,
+  String $backup_and_sync_cron_minute,
+  String $backup_and_sync_cron_month,
+  String $backup_and_sync_cron_monthday,
+  String $backup_and_sync_cron_weekday,
+  String $backup_and_sync_destination_host,
+  String $backup_and_sync_destination_path,
+  Boolean $backup_and_sync_enable,
+  String $grafana_admin_password,
+  String $grafana_admin_user,
+  String $ldap_sync_base_dir,
+  String $ldap_sync_cron_hour,
+  String $ldap_sync_cron_minute,
+  String $ldap_sync_cron_month,
+  String $ldap_sync_cron_monthday,
+  String $ldap_sync_cron_weekday,
+  Boolean $ldap_sync_enable,
+  Array[String] $ldap_sync_groups,
+  String $whitelabel_subtitle,
+  String $whitelabel_title,
+) {
+
+  file { '/root/grafana_tools':
+    ensure  => directory,
+    #notify  => Exec['ensure_grafana_tools_scripts_executable'],
+    recurse => true,
+    replace => false,
+    source  => "puppet:///modules/${module_name}/root/grafana_tools",
+  }
+
+  # chmod u+x grafana_tools scripts, but only when the file changes
+  exec { 'ensure_grafana_tools_scripts_executable':
+    command     => 'find /root/grafana_tools -name \'*.sh\' -exec chmod u+x {} \;',
+    path        => ['/usr/bin', '/usr/sbin'],
+    subscribe   => File['/root/grafana_tools'],
+    refreshonly => true,
+  }
+
+  # SETUP CONFIG FILES FOR VARIOUS grafana_tools
+  File_line {
+    ensure => 'present',
+    #replace => true,
+    require => File['/root/grafana_tools'],
+  }
+
+  # backup_and_sync/config
+  if ( ! empty($backup_and_sync_destination_host) ) {
+    file_line { 'backup_and_sync/config destination_host':
+      path  => '/root/grafana_tools/backup_and_sync/config',
+      line  => "destination_host=${backup_and_sync_destination_host}",
+      match => '^destination_host.*',
+    }
+  }
+  file_line { 'backup_and_sync/config destination_path':
```

Comment:
Was just curious the reasoning for updating the config files this way as opposed to making them as templates. The regex matching worries me a bit in that if we want to make new variables in the future we have to make sure no other regex's are going to clobber eachother.

Reply:
2 reasons:
1. The source files for this come from another git repo: https://github.com/jdmaloney/grafana_tools  I didn't really want to reimplement all of those files as templates.
2. I'm not experienced in using templates.

Reply:
Gotcha, was going to need to do something similar with a process monitoring script and was looking for ideas if one approach is better than another. Makes sense not to re-implement things now.

Reply:
I agree that templates should be avoided because it necessitates that this puppet class is the owner of the file and updates to that file can never be made from anywhere else.